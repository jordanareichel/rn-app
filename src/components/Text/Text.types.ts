export type TextProps = {
  fontSize: number;
  color: string;
  children: React.ReactNode;
  align?: 'left' | 'right' | 'center';
};
