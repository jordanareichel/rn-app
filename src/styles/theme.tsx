export const theme = {
  font: {
    sm: '12',
    md: '14',
    lg: '18',
  },
  colors: {
    dark: '#000000',
    light: '#ffffff',
  },
};
