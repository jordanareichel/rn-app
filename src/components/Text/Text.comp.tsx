import React from 'react';
import {TextRn} from './Text.styles';
import {TextProps} from './Text.types';

export const Text: React.FC<TextProps> = props => {
  const {children, ...rest} = props;

  return <TextRn {...Object.assign({...rest})}>{children}</TextRn>;
};

Text.defaultProps = {
  color: 'black',
  fontSize: 18,
  fontWeight: 'normal',
};
