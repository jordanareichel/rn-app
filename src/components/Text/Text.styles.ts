import styled, {css} from 'styled-components/native';
import {Text} from 'react-native';
import {TextProps} from './Text.types';

export const TextRn = styled(Text)<TextProps>`
  ${({fontSize = '', color = '', align = 'center'}) => css`
    font-size: ${fontSize}px;
    color: ${color};
    text-align: ${align};
  `}
`;
