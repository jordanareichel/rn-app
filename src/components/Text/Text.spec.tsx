import React from 'react';

import TestRenderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Text} from './Text.comp';
import {theme} from '@styles/theme';

describe('Component Text', () => {
  test('renders correctly', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <Text fontSize={18} fontWeight="bold">
          This is the Text
        </Text>
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
