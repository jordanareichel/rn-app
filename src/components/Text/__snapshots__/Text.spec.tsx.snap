// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Text renders correctly 1`] = `
<Text
  color="black"
  fontSize={18}
  fontWeight="bold"
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      },
    ]
  }
>
  This is the Text
</Text>
`;
